tensorboard/plugins/custom_scalar/custom_scalars_plugin.py

```diff
@@ -0,0 +1,274 @@
+    run: A run string to find tags for.
+    tag: A string that is a regex used to find matching tags.
+    The response is a JSON object:
+    {
+      // Whether the regular expression is valid. Also false if empty.
+      regexValid: boolean,
+
+      // An object mapping tag name to a list of ScalarEvents.
+      payload: Object<string, ScalarEvent[]>,
+    }
+    """
+    # TODO: return HTTP status code for malformed requests
+    tag_regex_string = request.args.get('tag')
+    run = request.args.get('run')
+
+    # TODO(chihuahua): Support OutputFormat.CSV via a 'format' GET param.
+    mime_type = 'application/json'
+
+    if not tag_regex_string:
+      # The user provided no regex.
+      return http_util.Respond(request, {
+          _REGEX_VALID_PROPERTY: False,
+          _TAG_TO_EVENTS_PROPERTY: {},
+      }, mime_type)
+
+    # Construct the regex.
+    try:
+      regex = re.compile(tag_regex_string)
+    except re.error:
+      return http_util.Respond(request, {
+          _REGEX_VALID_PROPERTY: False,
+          _TAG_TO_EVENTS_PROPERTY: {},
+      }, mime_type)
+
+    # Fetch the tags for the run. Filter for tags that match the regex.
+    run_to_data = self._multiplexer.PluginRunToTagToContent(
+        scalars_metadata.PLUGIN_NAME)
+
+    tag_to_data = None
+    try:
+      tag_to_data = run_to_data[run]
+    except KeyError:
+      # The run could not be found. Perhaps a configuration specified a run that
+      # TensorBoard has not read from disk yet.
+      payload = {}
+      tf.logging.warn('No scalar data found for run %r', run)
+
+    if tag_to_data:
+      scalars_plugin_instance = self._get_scalars_plugin()
+      if not scalars_plugin_instance:
+        return http_util.Respond(
+            request=request,
+            content=('Failed to respond to request for /scalars. '
+                     'The scalars plugin is oddly not registered.'),
+            content_type='text/plain',
+            code=500)
+
+      form = scalars_plugin.OutputFormat.JSON
+      payload = {tag: scalars_plugin_instance.scalars_impl(tag, run, form)[0]
+                 for tag in tag_to_data.keys()
+                 if regex.match(tag)}
+
+    # Produce the response.
+    return http_util.Respond(request, {
+        _REGEX_VALID_PROPERTY: True,
+        _TAG_TO_EVENTS_PROPERTY: payload,
+    }, mime_type)
+
+  @wrappers.Request.application
+  def layout_route(self, request):
+    """Fetches the custom layout specified by the config file in the logdir.
+
+    The response is a JSON object mirroring properties of the Layout proto:
+    {
+      categories: [
+        {
+          title: 'Biases',
+          regexes: ['foo/layer\d+/biases'],
+        },
+        {
+          title: 'Activations',
+          regexes: ['bar/layer0/biases', 'foo/layer0/biases'],
+        },
+      ]
+    }
+
+    The response is an empty object if no layout could be found.
+    """  # pylint: disable=anomalous-backslash-in-string
```

Comment:
…why? What's wrong with using a raw docstring (`r"""Fetches…"""`) or properly escaping the metacharacter (`\\d+/biases`)?

Reply:
Using a raw doc string sounds great! Done.